src/ray/core_worker/core_worker.cc

```diff
@@ -470,9 +470,7 @@ Status CoreWorker::Get(const std::vector<ObjectID> &ids, const int64_t timeout_m
       // If we got the result for this plasma ObjectID, the task that created it must
       // have finished. Therefore, we can safely remove its reference counting
       // dependencies.
-      if (!ids[i].IsDirectCallType()) {
-        RemoveObjectIDDependencies(ids[i]);
-      }
+      RemoveObjectIDDependencies(ids[i]);
```

Comment:
Oops, guess we need this for direct calls after all.

Reply:
could you please update the comment above also?